handler/core/src/test/java/io/knotx/fragments/handler/action/CircuitBreakerActionTest.java

```diff
@@ -134,10 +161,109 @@ void expectTimeout(VertxTestContext testContext, Vertx vertx) throws Throwable {
     }
   }
 
-  private static void apply(FragmentContext fragmentContext,
+  @Test
+  @DisplayName("Expect fallback transition when doAction ends up _error transition and circuit breaker is in fallback mode.")
+  void expectFallbackWhenDoActionEndsErrorTransition(VertxTestContext testContext, Vertx vertx) throws Throwable {
+    // given
+    CircuitBreakerOptions options = new CircuitBreakerOptions().setFallbackOnFailure(true);
+    CircuitBreaker circuitBreaker = new CircuitBreakerImpl("name", vertx, options);
+
+    CircuitBreakerAction tested = new CircuitBreakerAction(circuitBreaker,
+        CircuitBreakerActionTest::applyErrorTransition,"tested", INFO, ERROR_TRANSITION);
+
+    // when
+    tested.apply(new FragmentContext(FRAGMENT, new ClientRequest()),
+        testContext.succeeding(result -> {
+          testContext
+              .verify(() -> {
+                Assertions.assertEquals(FALLBACK_TRANSITION, result.getTransition());
+
+                ActionLog actionLog = new ActionLog(result.getNodeLog());
+                String fallback = actionLog.getLogs().getString("fallback");
+                List<ActionInvocationLog> doActionsLogs = actionLog.getDoActionLogs();
+
+                Assertions.assertEquals(1, doActionsLogs.size());
+                ActionInvocationLog invocationLog = doActionsLogs.get(0);
+
+                Assertions.assertFalse(invocationLog.isSuccess());
+                Assertions.assertNotNull(fallback);
+                Assertions.assertTrue(fallback.contains("DoActionExecuteException"));
```

Comment:
I understand this approach but what if:
- first invocation, timeout
- second invocation, timeout
- third invocation, the `_error` transition

I will get `DoActionExecuteException` but forget about first two reasons.

Reply:
hmm. But what exception is send depends only from CB implementation. I don't have any way to handle it each time exception is thrown. So I am taking what I get from vertx